### PR TITLE
Complete Budget App

### PR DESCRIPTION
**Pull request by principlebrothers**
## Complete Project Development
In this project I achieved the following;

- [x] Understood Rails RESTful design and router.
- [x] Understood Rails naming conventions.
- [x] Used preprocessed HTML file with embedded Ruby code.
- [x] Used params from browser requests in a safe way.
- [x] Used layouts and templates for shared content.
- [x] Use database migration files to maintain database schema.
- [x] Used validations for models.
- [x] Secure app from n+1 problems.
- [x] Used controllers to handle requests and render empty views.
- [x] Understood what ORM is.
- [x] Wrote SQL queries with ActiveRecord.
- [x] Set up associations between models.
- [x] Built a web app that requires the user to log in.
- [x] Described the difference between authorization and authentication.
- [x] Use Devise gem for authentication.
- [x] Understood how sessions and cookies can support authentication.
- [x] Limit access to web app resources based on authorization rules

[Video Demo Link](https://www.loom.com/share/0c24f68f0c4e417995036c5a306ccfc0)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,23 +220,21 @@ If you like this project...
 
 > Give credit to everyone who inspired your codebase.
```
Comment (1GORDON) [inline README.md]: Great job @principlebrothers 🤝

- [ ] As stated in the project requirements, kindly give appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).
- [ ] [Optional] Kindly fix the broken MIT license link to make your readme file more professional.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @principlebrothers ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] As stated in the project requirements, kindly record a video demonstrating all project features, and don't forget to highlight a piece of the code you find interesting or proud of making.
- [ ] You may add the link to this video presentation to your pull request.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (principlebrothers) [reply README.md]: Thanks @1GORDON for the feedback. I will implement the changes right away

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @principlebrothers 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Suggested changes

- [ ] Nice job :clap: on making changes suggested by the previous reviewer by recording a video presentation, however, the Reviewer said again that you should put the link of your video presentation in the Pull request description, kindly do that and your project will be ready go
image




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question  `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by principlebrothers** [comment]
Thanks for the review, @andersonlebon. But Added the video recording to the pull `README.md`.  I don't think this should prevent my progress if it is in the README.md.

**Review by KDlamini — APPROVED** [review]
### Hi @principlebrothers 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Great PR title and description
✔️ Models, controllers, routes, and view set up correctly
✔️ Application works as expected
✔️ Correct use of Gitflow
✔️ No linter errors

# Status: Approved ✔️ 

LeonardoInGatsby

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principlebrothers** [comment]
@KDlamini, thanks for the review.

**Pull request merged**